Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -1303,6 +1303,20 @@ public void RemoveDevice(InputDevice device, bool keepOnListOfAvailableDevices =
             // Let listeners know.
             for (var i = 0; i < m_DeviceChangeListeners.length; ++i)
                 m_DeviceChangeListeners[i](device, InputDeviceChange.Removed);
+
+            // For keyboard, try to set one as current.
+            // https://fogbugz.unity3d.com/f/cases/1305016/
+            for (var i = 0; i < m_Devices.Length; ++i)
+            {
+                var candidate = m_Devices[i];
+                if (candidate == null || !candidate.enabled)
+                    continue;
+
+                if (!(device is Keyboard) || !(candidate is Keyboard))
+                    continue;
+                candidate.MakeCurrent();
+                break;
+            }
```

Comment:
Still thinking about how we could solve this on a generic basis.

At least as far as `Keyboard` is concerned, though, this can more elegantly be solved by changing the code in `Keyboard.OnRemove`.

```
        /// <summary>
        /// Called when the keyboard is removed from the system.
        /// </summary>
        protected override void OnRemoved()
        {
            base.OnRemoved();
            if (current == this)
                current = null; // <------- Look for other keyboard here instead.
        }
```

Reply:
As far as doing this generically goes, I'm thinking the right way would be to pin this to `InputControlLayout.isGenericTypeOfDevice`. I.e. when a device is removed, the next device of the same generic type should be made current. So if you remove something that is a `Mouse`, the next device that is a `Mouse` is made current.

Problem is we don't want to load layouts here and we don't currently have a way of quickly determining the name of the first base layout that has `InputControlLayoutAttribute.isGenericTypeOfDevice` set. In other words, we'd first have to make this query quick and easy.

Reply:
`Mouse` inherits from `Pointer` and both `Mouse` and `Pointer` at a same time have `current`, so calling `device.MakeCurrent()` when it's a mouse will reset the `Pointer`. Because of this it's not a simple task of just traversing class hierarchies, code needs to understand semantics of where the `current` lives in types and the side effects of calling `MakeCurrent()`.

In this particular scenario I would like to focus on the actual bug: user just disables `OnScreenButton` and their `Keyboard.current` gets corrupted (by design). So we only need to fix it for `Keyboard`, and for `Keyboard` the heuristic of "set whatever first Keyboard you have because it will be the default system keyboard" makes a lot of sense. And after that we can figure out some functionality improvements later on (aka feature requests / refactoring / etc).